### PR TITLE
Add Corspman Access & Airlocks

### PR DESCRIPTION
**Pull request by rosieposieeee**
## About the PR

Content Client_AQ40yzRyFY

"""Corpsman""" access previously was just medical access on their locker (lol!) and NO access on their suit locker (LOL)

Tested and works. HoS was given the access too. Warden left out.

**Changelog**
:cl:
- add: Added Corpsman-only access and airlocks.

### CODE REVIEW

**Comment by DangerRevolution** [comment]
PR should say "Fixes #891" so the issue is autoclosed on merge :)

**Comment by rosieposieeee** [comment]
> PR should say "Fixes #891" so the issue is autoclosed on merge :)

ok lol i dont know how literal that is so i've just added that wholecloth

**Comment by DebugOk** [comment]
2 unrelated things in the same pr, no attempt at upstreaming it and they don't need a seperate access type. Medical + security is enough.

**Comment by DangerRevolution** [comment]
> they don't need a seperate access type. Medical + security is enough.

if a doctor gets into security they can just open Corpsman locker, as it's just a medical access locker? on the other way around, a security officer with medical access can just get into their locker. I've gotten Medical access a few times as Sec, doesn't mean I should be able to just go into the Corpsman's area and rob their shit

>2 unrelated things in the same pr, no attempt at upstreaming it

Wizden doesn't have Mantis, nor Corpsman

**Comment by IamVelcroboy** [comment]
> they don't need a seperate access type. Medical + security is enough.

I kinda disagree. I'm always going to push for better granular access types. The current med+sec access makes for some weird mapping setups on some maps where sec has access to corpsman's stuff or med has access to corpsman's room. This would allow for better control/less wonky mapping setups. 

Also, as pointed out, WD does not have a brig medic/corpsman role any more. It was removed.

**Comment by rosieposieeee** [comment]
image

it took a beautiful and vindicating 3 minutes for upstream to close it. 

image


> 2 unrelated things

its two airlocks.

**Comment by DebugOk** [comment]
I would argue with Sloth's logic but there really isn't a point

File: Resources/Prototypes/DeltaV/Entities/Structures/Doors/Airlocks/access.yml [hunk 1]
```diff
@@ -13,3 +13,27 @@
   components:
   - type: AccessReader
     access: [["Mantis"]]
+
+- type: entity
+  parent: AirlockMaintRnDLocked
```
Comment (DebugOk) [inline Resources/Prototypes/DeltaV/Entities/Structures/Doors/Airlocks/access.yml]: Unrelated to this pr

**Review by DebugOk — CHANGES_REQUESTED** [review]

**Review by DebugOk — CHANGES_REQUESTED** [review]
You also have to add the access here https://github.com/DeltaV-Station/Delta-v/blob/master/Content.Shared/Access/Components/IdCardConsoleComponent.cs#L44-L89 or no one can assign it

**Comment by DebugOk** [comment]
> You also have to add the access here https://github.com/DeltaV-Station/Delta-v/blob/master/Content.Shared/Access/Components/IdCardConsoleComponent.cs#L44-L89 or no one can assign it

and [here](https://github.com/DeltaV-Station/Delta-v/blob/master/Resources/Prototypes/Entities/Objects/Tools/access_configurator.yml#L20-L59)

**Comment by rosieposieeee** [comment]
gah, i always forget somewhere

**Review by DebugOk — APPROVED** [review]

**Pull request merged**